src/Microsoft.ML.Data/Transforms/TrainAndScoreTransform.cs

```diff
@@ -131,7 +152,36 @@ public sealed class Arguments : ArgumentsBase<SignatureTrainer>
 
         internal const string Summary = "Trains a predictor, or loads it from a file, and runs it on the data.";
 
+        /// <summary>
+        /// Convenience method for creating <see cref="TrainAndScoreTransform"/>.
+        /// The <see cref="TrainAndScoreTransform"/> allows for model stacking (i.e. to combine information from multiple predictive models to generate a new model)
+        /// in the pipeline by training a model first and then using the scores from the trained model.
+        ///
+        /// Unlike <see cref="ScoreTransform"/>, the <see cref="TrainAndScoreTransform"/> trains the model on the fly as name indicates.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>.</param>
+        /// <param name="trainer">The <see cref="ITrainer"/> object i.e. the learning algorithm that will be used for training the model.</param>
+        /// <param name="featureColumn">Role name for features.</param>
+        /// <param name="labelColumn">Role name for label.</param>
+        /// <returns></returns>
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, ITrainer trainer, string featureColumn = DefaultColumnNames.Features, string labelColumn = DefaultColumnNames.Label)
```

Comment:
>labelColumn [](start = 160, length = 11)

don't you need custom columns here?
also if you can split this long line into two, it would be nice.

Reply:
It can be but I see CustomColumns as advanced argument (not frequently used). Let see if other reviewers have same argument.

---
In reply to: [206365422](https://github.com/dotnet/machinelearning/pull/614#discussion_r206365422) [](ancestors = 206365422)

Reply:
well, it's not only CustomColumns, but also Weight, Group, Name, considering what they all have default values and user don't have to specify them, I don't see why not to add them, but I would also hear other people opinion. Just in case, same applies for Group column and CustomColumns in Scorer

---
In reply to: [206366117](https://github.com/dotnet/machinelearning/pull/614#discussion_r206366117) [](ancestors = 206366117,206365422)

Reply:
Let see if @TomFinley has any suggestion in this regard?

Reply:
Thank you for the question @zeahmed. My tendency would be to view group as something possibly useful. Weight and name, I view as less fundamentally useful. Custom columns, I am somewhat reluctant to support because I have some difficulty imagining how to explain that one to people. :P

However, I cannot quite claim to care that much. not because it isn't an important consideration but because the proposal of #581 renders it moot. We're going to be moving towards a world where, essentially, *all* trainers *become* TrainAndScore transforms, with more local decisions about what column roles are and aren't important.

---
In reply to: [206660387](https://github.com/dotnet/machinelearning/pull/614#discussion_r206660387) [](ancestors = 206660387)